core/templates/dev/head/services/UserService.ts

```diff
@@ -17,48 +17,56 @@
  */
 
 oppia.factory('UserService', [
-  '$http', '$q', '$window', 'UrlInterpolationService', 'UserInfoObjectFactory',
-  'DEFAULT_PROFILE_IMAGE_PATH',
-  function($http, $q, $window, UrlInterpolationService, UserInfoObjectFactory,
-      DEFAULT_PROFILE_IMAGE_PATH) {
+  '$http', '$q', '$window', 'UrlInterpolationService',
+  'UserInfoObjectFactory', 'DEFAULT_PROFILE_IMAGE_PATH',
+  function($http, $q, $window, UrlInterpolationService,
+      UserInfoObjectFactory, DEFAULT_PROFILE_IMAGE_PATH) {
     var PREFERENCES_DATA_URL = '/preferenceshandler/data';
 
     var userInfo = null;
 
     var getUserInfoAsync = function() {
-      if (GLOBALS.userIsLoggedIn) {
-        if (userInfo) {
-          return $q.resolve(userInfo);
-        }
-        return $http.get(
-          '/userinfohandler'
-        ).then(function(response) {
-          userInfo = UserInfoObjectFactory.createFromBackendDict(response.data);
-          return userInfo;
-        });
-      } else {
-        return $q.resolve(UserInfoObjectFactory.createDefault());
+      var currentUrl = $window.location.href;
+      if (currentUrl.includes('signup')) {
+        userInfo = UserInfoObjectFactory.createDefault();
+        return $q.resolve(userInfo);
+      }
+      if (userInfo) {
+        return $q.resolve(userInfo);
       }
+      return $http.get(
+        '/userinfohandler'
```

Comment:
Can we follow the patter live we do everywhere:
```
    return $http.get('/userinfohandler').then(function(response) {
      ....
```

Reply:
Done.